runtime/parachains/src/system_token_manager.rs

```diff
@@ -186,939 +123,936 @@ pub mod pallet {
+		min_balance: T::Balance,
+	) -> DispatchResult {
+		ensure!(
+			!OrigingalSystemTokenMetadata::<T>::contains_key(&original),
+			Error::<T>::OriginalAlreadyRegistered
+		);
+		let system_token_metadata = Self::system_token_metadata(&issuer, &description, &url)?;
+		let asset_metadata = Self::asset_metadata(&name, &symbol, decimals, min_balance)?;
+		let SystemTokenId { para_id, .. } = original.clone();
+		Self::try_push_sys_token_for_para_id(para_id, &original)?;
+		Self::try_push_para_id(para_id, &original)?;
+
+		SystemTokenProperties::<T>::insert(
+			&original,
+			SystemTokenProperty {
+				system_token_weight: Some(system_token_weight),
+				created_at: Self::unix_time(),
+			},
+		);
+		// Self registered as wrapped token, which would be used for knowing it is 'original system token'
+		WrappedSystemTokenOnPara::<T>::insert(&original, &original);
+
+		OrigingalSystemTokenMetadata::<T>::insert(
+			&original,
+			(&system_token_metadata, &asset_metadata),
+		);
+
+		Ok(())
+	}
+
+	/// **Description:**
+	///
+	/// Try register `wrapped_system_token` and return `weight` of system token
+	///
+	/// **Validity:**
+	///
+	/// - `SystemTokenId` is already registered.
+	///
+	/// - `WrappedSystemTokenId` is not registered yet.
+	///
+	/// - `ParaId` is not registered yet.
+	///
+	/// **Changes:**
+	///
+	/// - `WrappedSystemTokenOnPara`, `ParaIdSystemTokens`, `SystemTokenUsedParaIds`
+	fn try_register_wrapped(
+		original: &SystemTokenId,
+		wrapped: &SystemTokenId,
+	) -> Result<SystemTokenWeight, DispatchError> {
+		ensure!(
+			OrigingalSystemTokenMetadata::<T>::contains_key(original),
+			Error::<T>::OriginalNotRegistered
+		);
+		ensure!(
+			!WrappedSystemTokenOnPara::<T>::contains_key(wrapped),
+			Error::<T>::WrappedAlreadyRegistered
+		);
+
+		let SystemTokenId { para_id, .. } = wrapped.clone();
+
+		let para_ids = SystemTokenUsedParaIds::<T>::get(original)
+			.map_or(Default::default(), |para_id| para_id);
+		ensure!(!para_ids.contains(&para_id), Error::<T>::WrappedNotRegistered);
+
+		WrappedSystemTokenOnPara::<T>::insert(wrapped, original);
+		SystemTokenProperties::<T>::insert(
+			wrapped,
+			&SystemTokenProperty { system_token_weight: None, created_at: Self::unix_time() },
+		);
+
+		Self::try_push_sys_token_for_para_id(para_id, wrapped)?;
+		Self::try_push_para_id(para_id, original)?;
+
+		let property =
+			SystemTokenProperties::<T>::get(original).ok_or(Error::<T>::PropertyNotFound)?;
+		let weight = property.system_token_weight.map_or(BASE_SYSTEM_TOKEN_WEIGHT, |w| w);
+		Ok(weight)
+	}
+
+	/// **Description:**
+	///
+	/// Try deregister for all `original` and `wrapped` system tokens registered on runtime.
+	///
+	/// **Changes:**
+	///
+	///
+	fn try_deregister_all(original: SystemTokenId) -> DispatchResult {
+		let wrapped_system_tokens = Self::try_get_wrapped_system_token_list(&original)?;
+		for wrapped_system_token_id in wrapped_system_tokens {
+			Self::try_set_sufficient_and_unlink(wrapped_system_token_id, false)?;
```

Comment:
The original SystemTokenId is also included in the `token_list` because of `WrappedSystemTokenOnPara::<T>::insert(&original, &original); when executed register_system_token()`.

In case of `wrapped_system_token_id` == `original` `Self::try_set_sufficient_and_unlink(wrapped_system_token_id, false)?;` needs to be skipped. 
And `Self::try_deregister(SystemTokenType::Wrapped(wrapped_system_token_id, true))?;` still should be called due to removing `ParaIdSystemTokens` and `SystemTokenUsedParaIds`.

It was my fault from the first implemented `deregister` logic.

Reply:
But shouldn't we unlink the original token on original's local chain?

Reply:
I see your point, but still we need to set `sufficient` to `false`. That said, we should just change the state of `sufficient` for `wrapped_system_token_id == original`